torch/functional.py

```diff
@@ -989,16 +1009,38 @@ def tensordot(a, b, dims=2, out=None):
     """
     if has_torch_function_variadic(a, b):
         return handle_torch_function(tensordot, (a, b), a, b, dims=dims)
-    if isinstance(dims, (list, tuple)) or \
-       (isinstance(dims, torch.Tensor) and dims.numel() > 1):
+
+    dims_a: List[int] = []
+    dims_b: List[int] = []
+
+    if isinstance(dims, tuple):
         dims_a, dims_b = dims
-    else:
-        if isinstance(dims, torch.Tensor):
-            dims = dims.item()
+
+    if isinstance(dims, list):
+        assert len(dims) == 2
+        dims_a, dims_b = dims[0], dims[1]
+
+    if isinstance(dims, torch.Tensor):
+        num_elements = dims.numel()
+        if num_elements > 1:
+            assert dims.size()[0] == 2
+            dims_a = torch.jit.annotate(List[int], dims[0].tolist())
```

Comment:
Nit: not to use torch.jit.annotate use up to date syntax

Reply:
Unfortunately, this doesn't work at the moment. TS complains about redefining variable `dims_a`.

Reply:
file an issue and add in comment ?